protos/error.proto

```diff
@@ -0,0 +1,8 @@
+syntax = "proto3";
+
+// TODO to elaborate
+message Error {
```

Comment:
To nie jest najlepsza nazwa, od razu zaklada błąd, ja bym zrobil jakies ReturnMessage

Reply:
a czy mamy jeszcze jakąś inną informacje do przesłania?
poza tym w Go widzę, że się używa często-gęsto (prove me wrong)  if error != nil {}

Reply:
ale nie bedziemy uzywac tylko Go

```go
rpc CreateClass (ClassCreationRequest) returns (Error);
```

czytam to tak, ze funkcja po storzeniu zwraca bład, a akurat `CreateClass` albo inna funkcja powinna zwrocic ten sekretny kod ktory pozwoli na dolaczenie do zajec

Reply:
mogloby byc tak:
```go
ReturnMessage {
    message = "12345"
    code = OK(200)
    details = "Secret code for students"
}
```

Reply:
a dla bledu
```go
ReturnMessage {
    message = "Something went wrong"
    code = TEAPOT(418)
    details = "I'm a teapot"
}
```

Reply:
- w create class pasuje zwrócić jakiś odnośnik (id) stworzonych zajęć

Reply:
zatem moze
mogloby byc tak:
```go
CreateClassRepsonse {
    status = {
       // zamiast ustawiać jakies -1 w miejsce ID bedziemy sparwdzac czy status jest OK, jesli jest OK mozna wyslac sekretny kod do uczniow
        code = OK(200)
    }
    classId = 666
    secretCode = 12345
}
```

Reply:
okej, ale czemu aplikacje prezentera ma interesować w ogóle ten kod? 
czy nie lepiej aby sam serwer podczas wywołania prezentera _StartClass_  sam sobie go tworzył i notyfikował kogo trzeba?

Reply:
Do glowy przychodza mi 2 argumenty:

1. dlaczego prezenter ma nie miec informacji o tym ze istnieje jakis kod ktory powoduje ze do jego zajec mozna dolaczyc, co jesli nauczyciela z jakiegos powodu nie ma w sali, kto poda kod, mysle ze awaryjnie prezenter powinien miec informacje o kodzie dostepu do swoich zajec
2. jak to sie ma do Single Responsibility Principle? CreateClass ma tworzyc zajecia i wysylac kody do wszystkich nauczycieli?

Reply:
ok 1 mnie przekonuje
a co do 2 to _StartClass_ by się tym zajęło słaniem kodów